practice3.cpp

```diff
@@ -0,0 +1,312 @@
+    case 2: return Midfielder;
+    case 3: return Forward;
+    default:
+        cout << "Error!\n";
+        return Forward;
+    }
+}
+void OutputRole(Role role) {
+    switch (role) {
+    case 0: cout << "Goalkeeper"; break;
+    case 1: cout << "Defender"; break;
+    case 2: cout << "Midfielder"; break;
+    default: cout << "Forward"; break;
+    }
+}
+
+struct Footbolist {
+    string surname = "";
+    Role role = Forward;
+    short age = 0;
+    short games_count = 0;
+    int goal_count = 0;
+};
+Footbolist SetFootbolist() {
+    Footbolist footbolist;
+    cout << "\nFootbolist's surname: ";
+    cin >> (&footbolist)->surname;
+    (&footbolist)->role = SetRole();
+    cout << "Footbolist's age: ";
+    cin >> (&footbolist)->age;
+    cout << "Footbolist's games count: ";
+    cin >> (&footbolist)->games_count;
+    cout << "Footbolist's goal count: ";
+    cin >> (&footbolist)->goal_count;
+    return footbolist;
+}
+Footbolist SetOtherFootbolist() {
+    Footbolist footbolist;
+    cout << "New footbolist:\nFootbolist's surname: ";
+    cin >> (&footbolist)->surname;
+    (&footbolist)->role = SetRole();
+    cout << "Footbolist's age: ";
+    cin >> (&footbolist)->age;
+    cout << "Footbolist's games count: ";
+    cin >> (&footbolist)->games_count;
+    cout << "Footbolist's goal count: ";
+    cin >> (&footbolist)->goal_count;
+    return footbolist;
+}
+struct tnode {
+    Footbolist footbolist;
+    tnode* next = NULL;
+};
+typedef tnode* pnode;
+
+/////////////////////////////////////////////////////////////////////
+
+pnode formLIFO_1(int n) {
+    pnode head = NULL, cur;
+    Footbolist footbolist;
+    while (n > 0) {
+        footbolist = SetFootbolist();
+        n--;
+        cur = new tnode;
+        cur->footbolist = footbolist;
+        cur->next = head;
+        head = cur;
+    }
+    return head;
+}
+void formLIFO_2(int n, pnode * p) {
+    pnode head = NULL, cur;
+    Footbolist footbolist;
+    while (n > 0) {
+        footbolist = SetFootbolist();
+        n--;
+        cur = new tnode;
+        cur->footbolist = footbolist;
+        cur->next = head;
+        head = cur;
+    }
+    *p = head;
+}
+
+/////////////////////////////////////////////////////////////////////
+
+pnode formFIFO_1(int n) {
+    pnode head = NULL; 
+    pnode * cur = &head;
+    Footbolist footbolist;
+    while (n > 0) {
+        footbolist = SetFootbolist();
+        n--;
+        *cur = new tnode;
+        (*cur)->footbolist = footbolist;
+        cur = &(*cur)->next;
+    }
+    return head;
+}
+void formFIFO_2(int n, pnode * p) {
+    pnode head = NULL;
+    pnode* cur = &head;
+    Footbolist footbolist;
+    while (n > 0) {
+        footbolist = SetFootbolist();
+        n--;
+        *cur = new tnode;
+        (*cur)->footbolist = footbolist;
+        cur = &(*cur)->next;
+    }
+    *p = head;
+}
+
+/////////////////////////////////////////////////////////////////////
+bool deleteNode(pnode * head, pnode p) {
```

Comment:
would be better to notify user somehow that "football player" was successfully or not deleted. so that it would not look like your program went to an infinity loop.

Reply:
and same for other insert functions